Bottom: 1f7bea9a55e17a2547032e2851a8b113dcd2ec1d
Top:    0259758746426b71cd983a2073616ca4d4a81135
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-18 22:14:32 +0000

Remove dead colour selection code from nelma and gcode hids

(Silences some unused code warnings from clang)


---

diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 17bbae1..a3e1bfd 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -86,7 +86,6 @@ struct hid_gc_struct
   int width;
   unsigned char r, g, b;
   int erase;
-  int faded;
   struct color_struct *color;
   gdImagePtr brush;
 };
@@ -94,7 +93,6 @@ struct hid_gc_struct
 static struct color_struct *black = NULL, *white = NULL;
 static int linewidth = -1;
 static gdImagePtr lastbrush = (gdImagePtr)((void *) -1);
-static int lastcolor = -1;
 
 /* gd image and file for PNG export */
 static gdImagePtr gcode_im = NULL;
@@ -421,7 +419,6 @@ gcode_start_png_export ()
 
   linewidth = -1;
   lastbrush = (gdImagePtr)((void *) -1);
-  lastcolor = -1;
 
   hid_expose_callback (&gcode_hid, &region, 0);
 }
@@ -839,7 +836,6 @@ gcode_set_draw_xor (hidGC gc, int xor_)
 static void
 gcode_set_draw_faded (hidGC gc, int faded)
 {
-  gc->faded = faded;
 }
 
 static void
@@ -927,37 +923,6 @@ use_gc (hidGC gc)
       lastbrush = gc->brush;
 
     }
-#define CBLEND(gc) (((gc->r)<<24)|((gc->g)<<16)|((gc->b)<<8)|(gc->faded))
-  if (lastcolor != CBLEND (gc))
-    {
-      if (is_drill || is_mask)
-        {
-#ifdef FIXME
-          fprintf (f, "%d gray\n", gc->erase ? 0 : 1);
-#endif
-          lastcolor = 0;
-        }
-      else
-        {
-          double r, g, b;
-          r = gc->r;
-          g = gc->g;
-          b = gc->b;
-          if (gc->faded)
-            {
-              r = 0.8 * 255 + 0.2 * r;
-              g = 0.8 * 255 + 0.2 * g;
-              b = 0.8 * 255 + 0.2 * b;
-            }
-#ifdef FIXME
-          if (gc->r == gc->g && gc->g == gc->b)
-            fprintf (f, "%g gray\n", r / 255.0);
-          else
-            fprintf (f, "%g %g %g rgb\n", r / 255.0, g / 255.0, b / 255.0);
-#endif
-          lastcolor = CBLEND (gc);
-        }
-    }
 }
 
 static void
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index 9659b66..9e75927 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -102,7 +102,6 @@ struct hid_gc_struct {
 	Coord           width;
 	unsigned char   r, g, b;
 	int             erase;
-	int             faded;
 	struct color_struct *color;
 	gdImagePtr      brush;
 };
@@ -112,7 +111,6 @@ static HID nelma_hid;
 static struct color_struct *black = NULL, *white = NULL;
 static Coord    linewidth = -1;
 static gdImagePtr lastbrush = (gdImagePtr)((void *) -1);
-static int      lastcolor = -1;
 
 /* gd image and file for PNG export */
 static gdImagePtr nelma_im = NULL;
@@ -648,7 +646,6 @@ nelma_start_png_export()
 
 	linewidth = -1;
 	lastbrush = (gdImagePtr)((void *) -1);
-	lastcolor = -1;
 
 	hid_expose_callback(&nelma_hid, &region, 0);
 }
@@ -832,7 +829,6 @@ nelma_set_draw_xor(hidGC gc, int xor_)
 static void
 nelma_set_draw_faded(hidGC gc, int faded)
 {
-	gc->faded = faded;
 }
 
 static void
@@ -911,32 +907,6 @@ use_gc(hidGC gc)
 		lastbrush = gc->brush;
 
 	}
-#define CBLEND(gc) (((gc->r)<<24)|((gc->g)<<16)|((gc->b)<<8)|(gc->faded))
-	if (lastcolor != CBLEND(gc)) {
-		if (is_drill || is_mask) {
-#ifdef FIXME
-			fprintf(f, "%d gray\n", gc->erase ? 0 : 1);
-#endif
-			lastcolor = 0;
-		} else {
-			double          r, g, b;
-			r = gc->r;
-			g = gc->g;
-			b = gc->b;
-			if (gc->faded) {
-				r = 0.8 * 255 + 0.2 * r;
-				g = 0.8 * 255 + 0.2 * g;
-				b = 0.8 * 255 + 0.2 * b;
-			}
-#ifdef FIXME
-			if (gc->r == gc->g && gc->g == gc->b)
-				fprintf(f, "%g gray\n", r / 255.0);
-			else
-				fprintf(f, "%g %g %g rgb\n", r / 255.0, g / 255.0, b / 255.0);
-#endif
-			lastcolor = CBLEND(gc);
-		}
-	}
 }
 
 static void
